test/isolation.test.ts

```diff
@@ -0,0 +1,68 @@
+import { dirname, resolve } from 'path'
+import { Tinypool } from 'tinypool'
+import { fileURLToPath } from 'url'
+
+const __dirname = dirname(fileURLToPath(import.meta.url))
+
+test('idle workers can be recycled', async () => {
+  const pool = new Tinypool({
+    filename: resolve(__dirname, 'fixtures/sleep.js'),
+    minThreads: 4,
+    maxThreads: 4,
+    isolateWorkers: false,
+  })
+
+  function getThreadIds() {
+    return pool.threads.map((thread) => thread!.threadId).sort()
+  }
+
+  expect(pool.threads).toHaveLength(4)
+  const initialThreadIds = getThreadIds()
+
+  await Promise.all(times(4)(() => pool.run({})))
+  expect(getThreadIds()).toStrictEqual(initialThreadIds)
+
+  await pool.recycleWorkers()
+  expect(pool.threads).toHaveLength(4)
+
+  const newThreadIds = getThreadIds()
+  initialThreadIds.forEach((id) => expect(newThreadIds).not.toContain(id))
+
+  await Promise.all(times(4)(() => pool.run({})))
+  initialThreadIds.forEach((id) => expect(newThreadIds).not.toContain(id))
+  expect(getThreadIds()).toStrictEqual(newThreadIds)
+})
+
+test('running workers can recycle after task execution finishes', async () => {
+  const pool = new Tinypool({
+    filename: resolve(__dirname, 'fixtures/sleep.js'),
+    minThreads: 4,
+    maxThreads: 4,
+    isolateWorkers: false,
+  })
+
+  function getThreadIds() {
+    return pool.threads.map((thread) => thread!.threadId).sort()
+  }
+
+  expect(pool.threads).toHaveLength(4)
+  const initialThreadIds = getThreadIds()
+
+  const tasks = [
+    ...times(2)(() => pool.run({ time: 500 })),
+    ...times(2)(() => pool.run({ time: 2000 })),
+  ]
+
+  await Promise.all(tasks.slice(0, 2))
+  await pool.recycleWorkers()
```

Comment:
LGTM! Can we here have an expect for if the first two workers threadId changes here and the initialThreadIds don't have them anymore?

Reply:
Added checks here and some comments around them.